flink-formats/flink-hadoop-bulk/src/test/java/org/apache/flink/formats/hadoop/bulk/HadoopPathBasedPartFileWriterTest.java

```diff
@@ -82,13 +84,23 @@ public void testWriteFile() throws Exception {
         DataStream<String> stream =
                 env.addSource(new FiniteTestSource<>(data), TypeInformation.of(String.class));
         Configuration configuration = new Configuration();
-
+        // Elements from source  assign to one bucket , and produce two part after checkpoint.
```

Comment:
```suggestion
        // Elements from source are going to be assigned to one bucket and produce two output files after checkpoint
```
out of curiosity: what makes the output being split up into two files?

Reply:
In this test case use sync lock to make checkpoint twice , so fially we get two part file . Doc reference https://nightlies.apache.org/flink/flink-docs-master/docs/connectors/datastream/filesystem/#bucket-assignment
Code is in this method 
```
org.apache.flink.streaming.util.FiniteTestSource#run
```

Reply:
Thanks for the clarification. In that case, we might want to move the second part of the comment further up (see my related comment).